FFmpegInterop/FFmpegInteropMSS.cpp

```diff
@@ -1515,6 +1578,39 @@ void FFmpegInteropMSS::OnStarting(MediaStreamSource^ sender, MediaStreamSourceSt
 {
 	MediaStreamSourceStartingRequest^ request = args->Request;
 
+	if (isFirstSeek && avHardwareContext)
+	{
+		ID3D11Device* device = nullptr;
+		ID3D11DeviceContext* deviceContext = nullptr;
+
```

Comment:
Shouldn't we release those pointers?

Reply:
Frankly, the pointer handling is not very well done right now. We put both pointers into the hw device context. When the hw device context is released, the pointers are released. The better way would probably be to AddRef and Release them wherever we store them, that is how COM pointers are supposed to be used. Or maybe even better, we should be using ComPtr<>, which would automatically handle AddRef and Release for us.

Reply:
I tried with ComPtr, but it turned out tricky and the docs are very poor. In a lot of situations I was not sure how exactly to work with them. So I reverted and use classic AddRef/Release. When working with these COM interfaces, anyone taking ownership (store in a field) needs to AddRef and later Release. Now that I do it correctly in all placed, I don't see anymore crashes.